compiler/noirc_frontend/src/elaborator/patterns.rs

```diff
@@ -597,6 +600,28 @@ impl Elaborator<'_> {
         }
     }
 
+    // It's an error to call to the built-in function "warn" if we are not in a comptime context
+    fn check_call_to_warn_builtin_at_runtime(
```

Comment:
Is this not covered by how we disallow calling comptime functions from a runtime context?

Reply:
Ah, you are right. However, that check happens during monomorphization (not sure why!) and before that it panics when it can't find the built-in in SSA generation. That said, I guess I'll introduce this built-in in SSA that does nothing to let monomorphization produce this error.

Reply:
Hmm... but monomorphization happens before SSA, I don't know why it goes to SSA with errors 🤔

Reply:
Aah... it's because built-in functions don't go through the monomorphizer's queue.

Would it be okay to move this error from monomorphization to where I put it? (generalize it to all comptime functions, not just `warn`).

Reply:
I'm stopping for the day so we can discuss next week. You can workaround this with a non-builtin wrapper function which is comptime - this will ensure that the error is thrown appropriately.

Reply:
Good idea! And that explains why we have many wrappers like that.

But... it didn't work. The reason is that the warning is then produced in the std crate, and we don't show warnings if they aren't from the crate being compiled.

Reply:
Tbh that means that this feature isn't going to work for Nico's usage anyway

Reply:
Good point! I'll close this for now.

Reply:
@asterite we link macro errors back to the location which invokes the macro now. Seems like this means that these warnings would be visible now?

Reply:
Ah, but that won't work either. If a library has a macro that calls `warn` then those warnings will never be visible to callers of that library.